aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1317,6 +1317,28 @@ std::vector<Tensor> unsafe_split(const Tensor& self, int64_t split_size, int64_t
   return result;
 }
 
+std::vector<Tensor> hsplit(const Tensor& self, int64_t split_size, c10::optional<int64_t> dummy) {
+  TORCH_CHECK(!dummy.has_value(), "torch.hsplit expects dummy variable never take a value.")
+  TORCH_CHECK(self.dim() >= 1, "torch.hsplit requires a tensor with at least 1 dimension")
+  int64_t dim = (self.dim() == 1) ? 0 : 1;
+  TORCH_CHECK(self.sizes()[dim] % split_size == 0, "torch.hsplit's input axis : ", dim, " has dimension size : ", self.sizes()[dim], " which is not divisible by split_size=", split_size);
```

Comment:
There's no user-facing concept of an "input axis," so we can't reference here. We can do something like this, instead:

"torch.hsplit: attempted to split along dimension (dim), but the size of the dimension (dim size) is not divisible by the split_size (split_size)!"